code/modules/library/lib_items.dm

```diff
@@ -96,6 +96,44 @@
 		icon_state = "book-5"
 
 
+/obj/structure/bookcase/libraryspawn
+	var/spawn_category = "Any"
+	var/spawn_amount = 3
+
+/obj/structure/bookcase/libraryspawn/Initialize()
+	//list("Any", "Fiction", "Non-Fiction", "Adult", "Reference", "Religion")
+	.= ..()
+	spawn_category = sanitizeSQL(spawn_category)
+	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM ss13_library WHERE category=[spawn_category] ORDER BY RAND() LIMIT BY [spawn_amount]")

```

Comment:
Use prepared statements. Just in case.

```dm
	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM ss13_library WHERE category = :cat: ORDER BY RAND() LIMIT BY :amount:")
	query.Execute(list("cat" = spawn_category, "amount" = spawn_amount))
```

Reply:
As stated on Discord. `LIMIT BY` is erroneous syntax. It's just `LIMIT`. So the valid code would be:
```dm
	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM ss13_library WHERE category = :cat: ORDER BY RAND() LIMIT :amount:")
	query.Execute(list("cat" = spawn_category, "amount" = spawn_amount))
```

Reply:
oh and if doing prepared statements, drop the `sanitizeSQL` bit.